packages/venia-ui/lib/util/makeUrl.js

```diff
@@ -66,8 +67,8 @@ const makeOptimizedUrl = (path, { type, ...opts } = {}) => {
         }
     }
 
-    if (baseURL.href.startsWith(mediaBackend) && !useBackendForImgs) {
-        baseURL = new URL(baseURL.href.slice(baseURL.origin.length), origin);
+    if (baseURL.href.startsWith(magentoBackendURL) && !useBackendForImgs) {
+        baseURL = new URL(baseURL.href.slice(magentoBackendURL.length), origin);
```

Comment:
We talked offline about this but just to reiterate - the bug was that we set the base relative but we're only trimming the hostname such as `https://foo.com` even if you served from some sub path like `https://foo.com/venia`. Is that right?

I think I could use a comment here decoding this. :o Sorry for my obtuseness.

Reply:
Yes, good call. I'll add some comments to help explain what we're doing, and I'll also fixup and add tests so this file is fully covered.

Reply:
Added a test for relative URLs, and cleaned up the other tests that weren't accurately testing the functionality in c5b7c4b.

There is some hidden coupling between `makeUrl` and UPWARD config that we may want to re-visit. I'm going to create a task in the backlog to consider re-factoring, since two things have changed since it was originally written:
# GraphQL added full url fields to all image responses, so we no longer need to craft our own URLs based on image type.
# We have access to `secure_media_url`; UPWARD doesn't need to be hard-coded to `/media` anymore, we know exactly the pattern we should look for.